src/lib/FreeC/Pass/TypeInferencePass.hs

```diff
@@ -758,15 +811,18 @@ applyFuncDeclVisibly funcDecl = withLocalTypeAssumption $ do
 --   occur in the function declaration's type and on its right-hand side.
 --
 --   Fresh type variables used by the given type are replaced by regular type
---   varibales with the prefix 'freshTypeArgPrefix'. All other type variables
+--   variables with the prefix 'freshTypeArgPrefix'. All other type variables
 --   are not renamed.
-abstractTypeArgs :: [IR.QName] -> IR.FuncDecl -> IR.FuncDecl
-abstractTypeArgs typeArgs funcDecl =
-  let IR.TypeSchema _ _ typeExpr = fromJust (IR.funcDeclTypeSchema funcDecl)
-      (IR.TypeSchema _ typeArgs' _, subst) =
-          abstractTypeSchema' typeArgs typeExpr
-      funcDecl' = applySubst subst funcDecl
-  in  funcDecl' { IR.funcDeclTypeArgs = typeArgs' }
+abstractTypeArgs :: [IR.TypeVarIdent] -> IR.FuncDecl -> IR.FuncDecl
+abstractTypeArgs typeArgIdents funcDecl =
+  let
+    typeArgs                   = map (IR.UnQual . IR.Ident) typeArgIdents
+    IR.TypeSchema _ _ typeExpr = fromJust (IR.funcDeclTypeSchema funcDecl)
+    (IR.TypeSchema _ typeArgs' _, subst) =
+      abstractTypeSchema' typeArgs typeExpr
+    funcDecl' = applySubst subst funcDecl
+  in
+    funcDecl' { IR.funcDeclTypeArgs = typeArgs' }
```

Comment:
This suggestion cannot be applied directly since Brittany would change it back. I'm not exactly sure why Brittany wants to change the formatting in this case, though. Moving the binding of `typeArgs` down to the binding of `funcDecl'` seems to work. However, I usually try to sort `let`-bindings topologically. Thus, I'd stick with with the current formatting.